frontend/src/apis/feed.ts

```diff
@@ -1,8 +1,17 @@
 import requestAPI from './requestAPI';
 
+interface Feeds {
+  id: number;
+  name: string;
+  createdAt: string;
+  imageUrl: string;
+  likeCount: number;
+  content: string;
+}
+
 export const getFeeds = async () => {
```

Comment:
```suggestion
export const getFeeds = async (): Promise<Feeds[]> => {
```

위와 같이 표현하면 타입 단언을 쓰지 않고도 `data` 의 타입을 명시할 수 있을 거 같은데 혹시 타입 단언을 사용한 이유가 있을까요?

Reply:
```jsx
export const getFeeds = async (): Promise<Feeds[]> => {
  return await requestAPI.get('/feeds');
};
```

제가 다시 확인해 보니까 함수를 이렇게 쓰지 않으면 타입 추론이 안 되는 군요 🥹 개인적으로 타입 단언을 사용하면 타입 체크가 불가능하기 때문에 타입 선언으로 표현하는 게 좋을 거 같아서 한번 방법을 찾아 보겠습니다!

Reply:
```jsx
const apiMethods = async <T>({ method, endpoint, body }: ApiMethodsProps): Promise<T> => {
 ///
}

const requestAPI = {
  get: <T>(endpoint: string): Promise<T> => apiMethods({ method: 'GET', endpoint }),
 ///
}

export const getFeeds = async () => {
  const { data } = await requestAPI.get<{ data: Feeds[] }>('/feeds');
  return data;
};
```

만약 타입 단언을 쓰지 않는다면 위와 같이 제네릭으로 표현하는 방법이 있을 거 같은데 프룬은 어떤 방법이 낫다고 생각하시나요?

Reply:
타입 단언보다 제네릭을 사용하는 방법이 더 좋을 것 같아요! 짚어주셔서 감사해요😚

Reply:
제안해주신 제네릭이 더 적절할 것 같아서 리팩토링을 진행했습니다! 추가적으로 구조상 변경이 있었는데요, 제네릭을 추가하는 경우 아래와 같이 가독성이 좋지 않아서 `requestAPI` 내부에서 `apiMethods`, `get`, `post` 등의 http method를 호출하는 방식으로 개선했습니다.

```tsx
  get: <T>(endpoint: string): Promise<T> => apiMethods<T>({ method: 'GET', endpoint }),
  post: <T>(endpoint: string, body: Record<string, any>): Promise<T> =>
    apiMethods<T>({ method: 'POST', endpoint, body }),
  patch: <T>(endpoint: string, body: Record<string, any>): Promise<T> =>
    apiMethods<T>({ method: 'PATCH', endpoint, body }),
  delete: (endpoint: string): Promise<void> => apiMethods({ method: 'DELETE', endpoint })
```

```tsx
const requestAPI = {
  async apiMethods<T>({ method, endpoint, body }: ApiMethodsProps): Promise<T> {
    // ...
  },

  get<T>(endpoint: string): Promise<T> {
    return this.apiMethods<T>({ method: 'GET', endpoint });
  },
  // ...
}
```

Reply:
[a66d63f](https://github.com/christmas-tree-map/christmas-tree-map/pull/22/commits/a66d63f257266723990a5439d52f9555f99d2445)

Reply:
제 생각에도 body 타입을 좁힌다면 `Record<string, any>` 가 최선일 것 같습니다 🥲 리팩토링한 코드 너무 좋네요 👏